samplesheets/tests/test_permissions_api_taskflow.py

```diff
@@ -74,3 +81,489 @@ def test_get_archive(self):
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,
+            self.user_delegate_cat,
+            self.user_contributor_cat,
+            self.user_owner,
+            self.user_delegate,
+            self.user_contributor,
+        ]
+        bad_users = [
+            self.user_guest_cat,
+            self.user_finder_cat,
+            self.user_guest,
+            self.user_no_roles,
+        ]
+
+        self.assert_response_api(
+            self.url, good_users, 200, method='POST', data=self.post_data
+        )
+        self.assert_response_api(
+            self.url, bad_users, 403, method='POST', data=self.post_data
+        )
+        self.assert_response_api(
+            self.url, self.anonymous, 401, method='POST', data=self.post_data
+        )
+
+    @override_settings(PROJECTROLES_ALLOW_ANONYMOUS=True)
+    def test_create_anon(self):
+        """Test post() in IrodsRequestCreateAPIView with anonymous access"""
+        self.project.set_public()
+        self.assert_response_api(
+            self.url,
+            self.anonymous,
+            401,
+            method='POST',
+            data=self.post_data,
+        )
+
+    def test_create_archived(self):
+        """Test post() in IrodsRequestCreateAPIView with archived project"""
+        self.project.set_archive()
+        good_users = [self.superuser]
+        bad_users = [
+            self.user_owner_cat,
+            self.user_delegate_cat,
+            self.user_contributor_cat,
+            self.user_guest_cat,
+            self.user_finder_cat,
+            self.user_owner,
+            self.user_delegate,
+            self.user_contributor,
+            self.user_guest,
+            self.user_no_roles,
+        ]
+
+        self.assert_response_api(
+            self.url, good_users, 200, method='POST', data=self.post_data
+        )
+        self.assert_response_api(
+            self.url, bad_users, 403, method='POST', data=self.post_data
+        )
+        self.assert_response_api(
+            self.url, self.anonymous, 401, method='POST', data=self.post_data
+        )
+
+
+class TestIrodsRequestUpdateAPIView(TestIrodsRequestAPIViewBase):
+    """Test permissions for IrodsRequestUpdateAPIView"""
+
+    def setUp(self):
+        super().setUp()
+        self.update_data = {'path': self.path, 'description': 'Updated'}
+
+    def test_update(self):
+        """Test post() in IrodsRequestUpdateAPIView"""
+        good_users = [
+            self.superuser,
+            self.user_owner_cat,
+            self.user_delegate_cat,
+            self.user_owner,
+            self.user_delegate,
+        ]
+        bad_users = [
+            self.user_contributor_cat,
+            self.user_guest_cat,
+            self.user_finder_cat,
+            self.user_contributor,
+            self.user_guest,
+            self.user_no_roles,
+        ]
+
+        obj = self.create_request()
```

Comment:
I believe we only need to create the object once in `setUp()` for testing the permissions of an update view. Even if we update the model repeatedly with the same values, the response codes should still be the same.

The URL can also be set once in `self.url` inside `setUp()` in this case.